### DISCUSSION TITLE
Issues with windows Natural Voices and lexicons using gexgd0419's SAPI5 Tool

### DISCUSSION THREAD

**Discussion by johnysandels** [karashiiro/TextToTalk#223] [General]
I'm putting this here, because i feel like this may fall out of the scope of this project. and it doesn't really make sense to support voices that are using a third party program that could be patched out in future updates. Although the voices do sound really nice and it's free so I personally think it's worth looking into

I was testing the Natural voices from his voice adapter, and realized the voices weren't using lexicons. I tested the voices outside of ffxiv, and only could get the non online versions to actually work with lexicon files, I'm not sure if the Formatting for the xml is different, or if the lexicons need to be uploaded online, but i was running into issues. It seems like the online versions aren't reading the xml at all and are just reading the the basic text.

Also this is the Tool I'm referencing
https://github.com/gexgd0419/NaturalVoiceSAPIAdapter

**Comment by gexgd0419** [comment]
As the name suggests, this is just an "adapter", which builds a bridge between the SAPI5 interface and the Azure Speech SDK / Edge voice server WebSocket API.

This limitation is from the Edge voice server, which only allows [some of the SSML tags](https://github.com/gexgd0419/NaturalVoiceSAPIAdapter/wiki/Supported-SAPI-5-Features#limitation-when-using-microsoft-edge-online-voices).

The server rejects any SSML that has an unrecognized (disallowed) tag and no voice will be synthesized. So the adapter filters out all such tags before sending the text to the server. This way only the plain text will be read, but it's still better than nothing.

If you have an API key for the Azure Speech service, you can use Azure voices without this limitation (the v0.2 version supports an Azure API key).
